Manak_Leah_ndvi_2022.py

```diff
@@ -0,0 +1,670 @@
+
+student_ndvi_ts_single_site = _
+
+single_scene_points = 0
+
+# Ensure the data is stored in a dataframe.
+if isinstance(student_ndvi_ts_single_site, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    single_scene_points += 1
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_ts_single_site.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    single_scene_points += 2
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_ts_single_site.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# In[8]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# YOUR CODE HERE
+# Create an empty list with site, date, and mean ndvi information
+location_info = []
+
+# make a for loop for the location paths
+for site in site_paths:
+    # Get list of the location path with glob
+    locations = glob(os.path.join(site, "landsat-crop", "*"))
+    # grab the shapefiles from the locations and make the index 0
+    bounds = glob(os.path.join(site, "vector", "*-crop.shp"))[0]
+    # Open the shapefiles
+    opened_bound = gpd.read_file(bounds)
+
+    # Create a nested for loop for all locations
+    for all_locations in locations:
+        # Extract date and site info using "extract_date_sitename" function
+        site_info = extract_date_sitename(all_locations, [22, 26], [50, 58])
```

Comment:
@LManak for anyone who isn't familiar with this dataset, it might be helpful to comment on what these specific values mean